Sources/Atoms/Core/StoreContext.swift

```diff
@@ -201,11 +201,27 @@ internal struct StoreContext {
     }
 
     @usableFromInline
-    func reset(_ atom: some Atom) {
+    @_disfavoredOverload
+    func reset<Node: Atom>(_ atom: Node) {
+        let override = lookupOverride(of: atom)
+        let key = AtomKey(atom, overrideScopeKey: override?.scopeKey)
+
+        if let cache = lookupCache(of: atom, for: key) {
+            let newCache = makeNewCache(of: atom, for: key, override: override)
+            update(atom: atom, for: key, value: newCache.value, cache: cache, order: .newValue)
+        }
+    }
+
+    @usableFromInline
+    func reset<Node: Resettable>(_ atom: Node) {
         let override = lookupOverride(of: atom)
         let key = AtomKey(atom, overrideScopeKey: override?.scopeKey)
+        let state = getState(of: atom, for: key)
```

Comment:
This should be inside the condition as it's not in use outside of it.

Reply:
I updated the reset behavior, its straightforward and not checking `lookupCache(of: atom, for: key)`. 

Can you confirm